diff -ruN binutils-2.43/bfd/config.bfd binutils-test/bfd/config.bfd
--- binutils-2.43/bfd/config.bfd	2024-08-04 01:00:00.000000000 +0200
+++ binutils-test/bfd/config.bfd	2024-10-17 19:24:59.128885398 +0200
@@ -236,6 +236,18 @@
 
 case "${targ}" in
 # START OF targmatch.h
+  i[3-7]86-*-bunos*)
+    targ_defvec=i386_elf32_vec
+    targ_selvecs=
+    targ64_selvecs=x86_64_elf64_vec
+    ;;
+#ifdef BFD64
+  x86_64-*-bunos*)
+    targ_defvec=x86_64_elf64_vec
+    targ_selvecs=i386_elf32_vec
+    want64=true
+    ;;
+#endif
 #ifdef BFD64
   aarch64-*-darwin*)
     targ_defvec=aarch64_mach_o_vec
diff -ruN binutils-2.43/config.sub binutils-test/config.sub
--- binutils-2.43/config.sub	2024-08-04 01:00:00.000000000 +0200
+++ binutils-test/config.sub	2024-10-17 19:24:59.128885398 +0200
@@ -1768,7 +1768,7 @@
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* | mlibc* | cos* | mbr* | ironclad* )
+	     | fiwix* | mlibc* | cos* | mbr* | ironclad* | bunos*)
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
diff -ruN binutils-2.43/gas/configure.tgt binutils-test/gas/configure.tgt
--- binutils-2.43/gas/configure.tgt	2024-08-04 01:00:00.000000000 +0200
+++ binutils-test/gas/configure.tgt	2024-10-17 19:24:59.128885398 +0200
@@ -126,6 +126,7 @@
 generic_target=${cpu_type}-$vendor-$os
 # Note: This table is alpha-sorted, please try to keep it that way.
 case ${generic_target} in
+  i386-*-bunos*) fmt=elf ;;
   aarch64*-*-elf*)			fmt=elf;;
   aarch64*-*-fuchsia*)			fmt=elf;;
   aarch64*-*-haiku*)			fmt=elf em=haiku ;;
diff -ruN binutils-2.43/ld/configure.tgt binutils-test/ld/configure.tgt
--- binutils-2.43/ld/configure.tgt	2024-08-04 01:00:00.000000000 +0200
+++ binutils-test/ld/configure.tgt	2024-10-17 19:24:59.128885398 +0200
@@ -82,6 +82,16 @@
 # Please try to keep this table more or less in alphabetic order - it
 # makes it much easier to lookup a specific archictecture.
 case "${targ}" in
+i[3-7]86-*-bunos*)
+			targ_emul=elf_i386_bunos
+			targ_extra_emuls=elf_i386
+			targ64_extra_emuls="elf_x86_64_bunos elf_x86_64"
+			;;
+x86_64-*-bunos*)
+			targ_emul=elf_x86_64_bunos
+			targ_extra_emuls="elf_i386_bunos elf_x86_64 elf_i386"
+			;;
+
 aarch64_be-*-elf)	targ_emul=aarch64elfb
 			targ_extra_emuls="aarch64elf aarch64elf32 aarch64elf32b armelfb armelf"
 			;;
diff -ruN binutils-2.43/ld/emulparams/elf_i386_bunos.sh binutils-test/ld/emulparams/elf_i386_bunos.sh
--- binutils-2.43/ld/emulparams/elf_i386_bunos.sh	1970-01-01 01:00:00.000000000 +0100
+++ binutils-test/ld/emulparams/elf_i386_bunos.sh	2024-10-17 19:24:59.128885398 +0200
@@ -0,0 +1,3 @@
+. ${srcdir}/emulparams/elf_i386.sh
+GENERATE_SHLIB_SCRIPT=yes
+GENERATE_PIE_SCRIPT=yes
\ No newline at end of file
diff -ruN binutils-2.43/ld/emulparams/elf_x86_64_bunos.sh binutils-test/ld/emulparams/elf_x86_64_bunos.sh
--- binutils-2.43/ld/emulparams/elf_x86_64_bunos.sh	1970-01-01 01:00:00.000000000 +0100
+++ binutils-test/ld/emulparams/elf_x86_64_bunos.sh	2024-10-17 19:24:59.128885398 +0200
@@ -0,0 +1 @@
+. ${srcdir}/emulparams/elf_x86_64.sh
\ No newline at end of file
diff -ruN binutils-2.43/ld/Makefile.am binutils-test/ld/Makefile.am
--- binutils-2.43/ld/Makefile.am	2024-08-04 01:00:00.000000000 +0200
+++ binutils-test/ld/Makefile.am	2024-10-17 19:24:59.128885398 +0200
@@ -277,6 +277,7 @@
 	eelf32xtensa.c \
 	eelf32z80.c \
 	eelf_i386.c \
+	eelf_i386_bunos.c \
 	eelf_i386_be.c \
 	eelf_i386_fbsd.c \
 	eelf_i386_haiku.c \
@@ -461,6 +462,7 @@
 	eelf64tilegx_be.c \
 	eelf_mipsel_haiku.c \
 	eelf_x86_64.c \
+	eelf_x86_64_bunos.c \
 	eelf_x86_64_cloudabi.c \
 	eelf_x86_64_fbsd.c \
 	eelf_x86_64_haiku.c \
