diff -ruN gcc-14.2.0/config.sub gcc-patches/config.sub
--- gcc-14.2.0/config.sub	2024-08-01 10:17:13.000000000 +0200
+++ gcc-patches/config.sub	2024-10-17 19:29:07.813445132 +0200
@@ -1749,7 +1749,7 @@
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* )
+	     | fiwix* | bunos* )
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
diff -ruN gcc-14.2.0/fixincludes/mkfixinc.sh gcc-patches/fixincludes/mkfixinc.sh
--- gcc-14.2.0/fixincludes/mkfixinc.sh	2024-08-01 10:17:13.000000000 +0200
+++ gcc-patches/fixincludes/mkfixinc.sh	2024-10-17 19:27:52.333128562 +0200
@@ -11,6 +11,7 @@
 
 # Check for special fix rules for particular targets
 case $machine in
+    *-bunos* | \
     i?86-*-cygwin* | \
     i?86-*-mingw32* | \
     x86_64-*-mingw32* | \
diff -ruN gcc-14.2.0/gcc/config/bunos.h gcc-patches/gcc/config/bunos.h
--- gcc-14.2.0/gcc/config/bunos.h	1970-01-01 01:00:00.000000000 +0100
+++ gcc-patches/gcc/config/bunos.h	2024-10-17 19:27:52.333128562 +0200
@@ -0,0 +1,21 @@
+#undef TARGET_MYOS
+#define TARGET_MYOS 1
+#undef LIB_SPEC
+#define LIB_SPEC "-lc"
+
+#undef STARTFILE_SPEC
+#define STARTFILE_SPEC "crt0.o%s crti.o%s crtbegin.o%s"
+
+#undef ENDFILE_SPEC
+#define ENDFILE_SPEC "crtend.o%s crtn.o%s"
+
+#undef TARGET_OS_CPP_BUILTINS
+#define TARGET_OS_CPP_BUILTINS()      \
+ do {                                 \
+    builtin_define ("__bunos__");     \
+    builtin_define ("__unix__");      \
+    builtin_assert ("system=bunos");   \
+    builtin_assert ("system=unix");   \
+    builtin_assert ("system=posix");  \
+  } while(0);
+
diff -ruN gcc-14.2.0/gcc/config.gcc gcc-patches/gcc/config.gcc
--- gcc-14.2.0/gcc/config.gcc	2024-08-01 10:17:14.000000000 +0200
+++ gcc-patches/gcc/config.gcc	2024-10-17 19:27:52.333128562 +0200
@@ -722,6 +722,12 @@
 
 # Common parts for widely ported systems.
 case ${target} in
+*-*-bunos*)
+  gas=yes
+  gnu_ld=yes
+  default_use_cxa_atexit=yes
+  use_gcc_stdint=wrap
+  ;;
 *-*-darwin*)
   tmake_file="t-darwin "
   tm_file="${tm_file} darwin.h"
@@ -1189,6 +1195,12 @@
 esac
 
 case ${target} in
+i[34567]86-*-bunos*)
+    tm_file="${tm_file} i386/unix.h i386/att.h elfos.h glibc-stdint.h i386/i386elf.h bunos.h"
+    ;;
+x86_64-*-bunos*)
+    tm_file="${tm_file} i386/unix.h i386/att.h elfos.h glibc-stdint.h i386/i386elf.h i386/x86-64.h bunos.h"
+    ;;
 aarch64*-*-elf | aarch64*-*-fuchsia* | aarch64*-*-rtems*)
 	tm_file="${tm_file} elfos.h newlib-stdint.h"
 	tm_file="${tm_file} aarch64/aarch64-elf.h aarch64/aarch64-errata.h aarch64/aarch64-elf-raw.h"
diff -ruN gcc-14.2.0/libgcc/config.host gcc-patches/libgcc/config.host
--- gcc-14.2.0/libgcc/config.host	2024-08-01 10:17:17.000000000 +0200
+++ gcc-patches/libgcc/config.host	2024-10-17 19:27:52.336461911 +0200
@@ -409,6 +409,10 @@
 esac
 
 case ${host} in
+i[3-7]86-*-bunos* | x86_64-*-bunos*)
+	extra_parts="$extra_parts crti.o crtbegin.o crtend.o crtn.o"
+	tmake_file="$tmake_file i386/t-crtstuff t-crtstuff-pic t-libgcc-pic"
+	;;
 aarch64*-*-elf | aarch64*-*-rtems*)
 	extra_parts="$extra_parts crtbegin.o crtend.o crti.o crtn.o"
 	extra_parts="$extra_parts crtfastmath.o"
diff -ruN gcc-14.2.0/libstdc++-v3/crossconfig.m4 gcc-patches/libstdc++-v3/crossconfig.m4
--- gcc-14.2.0/libstdc++-v3/crossconfig.m4	2024-08-01 10:17:18.000000000 +0200
+++ gcc-patches/libstdc++-v3/crossconfig.m4	2024-10-17 19:27:52.336461911 +0200
@@ -8,6 +8,12 @@
   arm*-*-symbianelf*)
     # This is a freestanding configuration; there is nothing to do here.
     ;;
+  *-bunos*)
+    GLIBCXX_CHECK_COMPILER_FEATURES
+    GLIBCXX_CHECK_LINKER_FEATURES
+    GLIBCXX_CHECK_MATH_SUPPORT
+    GLIBCXX_CHECK_STDLIB_SUPPORT
+    ;;    
 
   avr*-*-*)
     AC_DEFINE(HAVE_ACOSF)
@@ -198,6 +204,8 @@
     AC_CHECK_FUNCS(_wfopen)
     GCC_CHECK_TLS
     ;;
+  *-myos*)
+    ;;
   *-netbsd* | *-openbsd*)
     SECTION_FLAGS='-ffunction-sections -fdata-sections'
     AC_SUBST(SECTION_FLAGS) 
